### PR TITLE
convert_model() in openvino.runtime.

### PR DESCRIPTION
**Pull request by popovaan**
Description: 
Added convert_model() in openvino.runtime.
convert_model() in openvino.runtime uses only new frontends, legacy functionality is not present.

Depends on: https://github.com/openvinotoolkit/openvino/pull/18029

Ticket: 99000

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

### CODE REVIEW

**Review by rkazants — COMMENTED** [review]
Please update CODEOWNERS file and label tags for the component.

Also please move TF FE layer tests to the new space. I think all of them should use TF FE

**Review by rkazants — COMMENTED** [review]
Do we have tests where we install openvino wheel, try to import `convert_model` and convert model?

File: src/bindings/python/src/openvino/runtime/__init__.py
Comment (rkazants) [inline src/bindings/python/src/openvino/runtime/__init__.py:71]: I think we should put `convert_model` and other relative stuff into the separate directory. Now we are mixing mo stuff with ops, etc. Don't think this is good.
Moreover, it will have to split our space in codeowners file instead to review all runtime stuff.

File: src/bindings/python/src/openvino/runtime/convert.py
Comment (rkazants) [inline src/bindings/python/src/openvino/runtime/convert.py:83]: I think we should not have legacy FE parameters here, also different legacy options like `transformation` config files.

File: tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py
Comment (rkazants) [inline tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py:35]: please delete this because I removed WA based on `convert_to_pb`.

File: src/bindings/python/src/openvino/runtime/convert_impl.py
Comment (rkazants) [inline src/bindings/python/src/openvino/runtime/convert_impl.py:275]: I think there should not be fallback to legacy TF FE

File: src/bindings/python/src/openvino/runtime/convert_impl.py
Comment (rkazants) [inline src/bindings/python/src/openvino/runtime/convert_impl.py:342]: no fallback

**Review by slyalin — CHANGES_REQUESTED** [review]

File: tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py
Comment (popovaan) [reply tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py]: Done.

File: src/bindings/python/src/openvino/runtime/convert_impl.py
Comment (popovaan) [reply src/bindings/python/src/openvino/runtime/convert_impl.py]: Yes, if fallback is needed we get the following error:
"The provided model cannot be converted with new frontend, as fallback to legacy is needed. Please try to install openvino-dev and use convert_model() from openvino.tools.mo."

File: src/bindings/python/src/openvino/runtime/convert_impl.py
Comment (popovaan) [reply src/bindings/python/src/openvino/runtime/convert_impl.py]: Replied in previous comment.

**Comment by popovaan** [comment]
> Do we have tests where we install openvino wheel, try to import `convert_model` and convert model?

Yes, MO unit tests are tested after installing of wheel package. 
We have unit tests where convert_model is imported and basic conversion is performed. 
[tools/mo/unit_tests/mo/convert/import_from_mo_test.py](https://github.com/openvinotoolkit/openvino/pull/18080/files#diff-18a84814676d7e09007f9dcdb55843133eff718760e4229e3a4413857c7676b7)

File: src/bindings/python/src/openvino/runtime/__init__.py
Comment (popovaan) [reply src/bindings/python/src/openvino/runtime/__init__.py]: Done.

File: tools/mo/automation/package_BOM.txt
Comment (rkazants) [inline tools/mo/automation/package_BOM.txt:41]: can we have package BOM testing for new `convert_model`? Will it be helpful?

**Review by rkazants — COMMENTED** [review]
Am I correct that MO unit-tests are not testing `convert_model` from `openvino.runtime`? Can we start to deploy separate MO unit-tests for new `convert_model` and `ovc`? And move tests that are easily moved?

File: tools/mo/automation/package_BOM.txt
Comment (ilya-lavrenov) [reply tools/mo/automation/package_BOM.txt:41]: do we need BOM at all?

Comment (rkazants) [reply tools/mo/automation/package_BOM.txt:41]: It checks a license at least

Comment (popovaan) [reply tools/mo/automation/package_BOM.txt:41]: Done.

**Comment by popovaan** [comment]
> Am I correct that MO unit-tests are not testing `convert_model` from `openvino.runtime`? Can we start to deploy separate MO unit-tests for new `convert_model` and `ovc`? And move tests that are easily moved?

Added unit tests which don't depend on legacy functionality to tools/ovc.

File: src/bindings/python/src/openvino/runtime/__init__.py
Comment (popovaan) [reply src/bindings/python/src/openvino/runtime/__init__.py:74]: As discussed offline exception will break import of convert_model from tools.ovc and there group of tests which need changes if add exception here. 
So I suggest to implement it in separate PR.

File: tools/CMakeLists.txt
Comment (rkazants) [inline tools/CMakeLists.txt:45]:
```suggestion
# OpenVino Conversion Tool
```

File: tools/mo/openvino/tools/mo/__init__.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/__init__.py:4]: can you please import from the full path: `openvino.tools.mo.convert`?

File: tools/mo/openvino/tools/mo/convert_impl.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/convert_impl.py:956]: why is this changed?

File: tools/mo/openvino/tools/mo/load/caffe/loader.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/load/caffe/loader.py:50]: why is this changed?

File: tools/mo/unit_tests/mo/convert/import_from_mo_test.py
Comment (rkazants) [inline tools/mo/unit_tests/mo/convert/import_from_mo_test.py:88]: Why do we test ovc `convert_model` here? Can we move this file to OVC unit-tests?

File: tools/mo/unit_tests/mo/convert/logger_test_actual.py
Comment (rkazants) [inline tools/mo/unit_tests/mo/convert/logger_test_actual.py:29]: the same question. May be we should leave this test without changes? And move it to OVC unit-tests after?

File: tools/mo/unit_tests/mo/convert/meta_data_test.py
Comment (rkazants) [inline tools/mo/unit_tests/mo/convert/meta_data_test.py:65]: why is this changed? I propose do not change MO unit-tests unless they are move to OVC unit-tests.

File: tools/ovc/openvino/tools/ovc/convert_impl.py
Comment (rkazants) [inline tools/ovc/openvino/tools/ovc/convert_impl.py:351]: please create a ticket and mention it in todo. It is important because I do not know what will be in case of `ovc` and fallback.

File: tools/ovc/unit_tests/moc_tf_fe/check_info_messages_test.py
Comment (rkazants) [inline tools/ovc/unit_tests/moc_tf_fe/check_info_messages_test.py:1]: did you remove them from MO unit-tests? May be it makes sense?

File: tools/ovc/unit_tests/ovc/utils/args_to_string_test.py
Comment (rkazants) [inline tools/ovc/unit_tests/ovc/utils/args_to_string_test.py:1]: may we should need to remove them from MO unit-test scope?

File: tools/ovc/unit_tests/ovc/utils/args_to_string_test.py
Comment (rkazants) [inline tools/ovc/unit_tests/ovc/utils/args_to_string_test.py:3]: do we have any tests for `ovc` tool? And `mo` tool?

File: tools/mo/openvino/tools/mo/convert_impl.py
Comment (popovaan) [reply tools/mo/openvino/tools/mo/convert_impl.py:956]: It looks like we have the wrong change on master, so I uncommented this code. If we intentionally commented it I suggest to change to `raise e` and remove the commented code.

File: tools/mo/openvino/tools/mo/load/caffe/loader.py
Comment (popovaan) [reply tools/mo/openvino/tools/mo/load/caffe/loader.py:50]: It was changed to remove the default value of "k" and "caffe_parser_path" from convert_model() definition.
If we leave it in definition, then for tools.ovc.convert_model() we would need to search location of legacy convert_model and set these legacy paths. 
So I changed default values for  "k" and "caffe_parser_path" to None and added setting the default paths in in places when they are actually used.

File: tools/mo/unit_tests/mo/convert/import_from_mo_test.py
Comment (popovaan) [reply tools/mo/unit_tests/mo/convert/import_from_mo_test.py:88]: This change was done before I added unit tests to ovc, so I can remove this change now.

File: tools/mo/unit_tests/mo/convert/meta_data_test.py
Comment (popovaan) [reply tools/mo/unit_tests/mo/convert/meta_data_test.py:65]: It was changed because I changed the import to ovc.convert_model which does not have MO_version, only runtime version. It looks redundant to have both, as they are in the same package.

As now we have unit tests in ovc I can remove this change.

File: tools/ovc/unit_tests/ovc/utils/args_to_string_test.py
Comment (popovaan) [reply tools/ovc/unit_tests/ovc/utils/args_to_string_test.py]: Done.

File: tools/ovc/unit_tests/moc_tf_fe/check_info_messages_test.py
Comment (popovaan) [reply tools/ovc/unit_tests/moc_tf_fe/check_info_messages_test.py:1]: I removed all check_info_messages tests from MO unit tests except for TestInfoMessagesTFFEWithFallback, as it checks different message. 
In ovc I added similar test that checks suggestion to install openvino-dev and use convert_model from tools.mo.

File: tools/mo/unit_tests/mo/convert/logger_test_actual.py
Comment (popovaan) [reply tools/mo/unit_tests/mo/convert/logger_test_actual.py]: Removed this change.

File: tools/ovc/unit_tests/ovc/utils/args_to_string_test.py
Comment (popovaan) [reply tools/ovc/unit_tests/ovc/utils/args_to_string_test.py]: Added `ovc` tool test tests/layer_tests/mo_python_api_tests/test_ovc_cli_tool.py
`mo` tool currently tested in `mo_args` tests and in some tests from MO Python API group (LegacyExtTest, TestSubprocessMoConvert)

**Review by rkazants — APPROVED** [review]
Looks good to me. Let us double-check that all test are in place in the pre-commit before merge

File: tools/mo/automation/package_BOM.txt
Comment (ilya-lavrenov) [inline tools/mo/automation/package_BOM.txt:41]: why do we need this file? is not it outdated?

File: tools/ovc/automation/create_package.py
Comment (ilya-lavrenov) [inline tools/ovc/automation/create_package.py:16]: why do we need this file? who is an user of `create_package.py`?

File: tools/mo/automation/package_BOM.txt
Comment (popovaan) [reply tools/mo/automation/package_BOM.txt:41]: This file is a part of BOM test. BOM test checks license at the head of each code file. So it looks like we still need it.

File: tools/ovc/automation/create_package.py
Comment (popovaan) [reply tools/ovc/automation/create_package.py]: Looks like there are no usages. I removed this file and moved package_BOM.txt to unit_tests.

**Comment by popovaan** [comment]
> Leftovers for the next PR:
> 
> 1. avoid `try-catch` in `openvino.runtime.init`. All internal py files should use internal `.py` files for importing OV stuff
> 2. move all possible unit-tests to ovc space
> 3. add PyLint checker for OVC sources and to azure
> 4. fix a gap for telemetry in ovc
> 5. fallback for TF FE??
> 
> Best regards, Roman

Telemetry is already added in this PR. 
The 'requests' dependency was removed from telemetry so we can now add dependency on telemetry to runtime.

For other tasks I created the ticket: 114255

**Pull request merged**
